build/common/nginx-default.conf.template

```diff
@@ -52,8 +52,8 @@ server {
     location @webserver {
         proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
         proxy_set_header X-Forwarded-Proto $scheme;
-        proxy_set_header X-Frappe-Site-Name $http_host;
-        proxy_set_header Host $host;
+        proxy_set_header X-Frappe-Site-Name $host;
+        proxy_set_header Host $http_host;
```

Comment:
https://github.com/frappe/bench/blob/develop/bench/config/templates/nginx.conf#L91

Reply:
## Change 1
In the bench template I see `proxy_set_header X-Frappe-Site-Name {{ site_name }};`.

**X-Frappe-Site-Name** to `$host` is necessary because it always only contains the host name and never the port number, whereas $http_host contains whatever is set by the client in the HTTP Host header and  it can contain port number as well. If you have to follow bench template then `$host` is the right usage here for **X-Frappe-Site-Name**.

## Change 2
For the change in the line `proxy_set_header Host $http_host;`, it is not needed really. It can remain the same as the previous one which is `$host`. But I thought it will be better to set the **Host** header with the value whatever was passed from the client. `$http_host` contains the value of the **Host** header that was set by the client. You can check about the same of using `$http_host` vs `$host` to set Host header in this [issue](https://github.com/nginx-proxy/nginx-proxy/issues/763)


I hope the above explanation  answers your question.

Reply:
for change 2 i would say you should've let it be. it's better to not let user send the input as to what the `$http_host` would be is not known, but `$host` always gets set to `$server_name`, so you'd never face any issue regarding malformed requests sent to the server